packages/xo-web/src/common/custom-fields.js

```diff
@@ -0,0 +1,149 @@
+import _ from 'intl'
+import defined from '@xen-orchestra/defined'
+import React from 'react'
+import PropTypes from 'prop-types'
+import { injectState, provideState } from 'reaclette'
+
+import ActionButton from './action-button'
+import Component from './base-component'
+import decorate from './apply-decorators'
+import Icon from './icon'
+import SingleLineRow from './single-line-row'
+import Tooltip from './tooltip'
+import { addCustomField, deleteCustomField, setCustomField } from './xo'
+import { confirm } from './modal'
+import { connectStore, noop } from './utils'
+import { Container, Col, Row } from './grid'
+import { createGetObject } from './selectors'
+import { error } from './notification'
+import { Text } from './editable'
+
+const CUSTOM_FIELDS_KEY = 'XenCenter.CustomFields.'
+
+class AddCustomFieldModal extends Component {
+  get value() {
+    return defined(this.state, {})
+  }
+
+  render() {
+    const { name, value } = this.state
+    return (
+      <div>
+        <SingleLineRow>
+          <Col size={6}>{_('name')}</Col>
+          <Col size={6}>
+            <input
+              autoFocus
+              className='form-control'
+              onChange={this.linkState('name')}
+              type='text'
+              value={name}
+            />
+          </Col>
+        </SingleLineRow>
+        <div className='mt-1'>
+          <SingleLineRow>
+            <Col size={6}>{_('value')}</Col>
+            <Col size={6}>
+              <input
+                className='form-control'
+                onChange={this.linkState('value')}
+                type='text'
+                value={value}
+              />
```

Comment:
Would it be possible to add a addon button to this field to enter a date easily?

Reply:
Not easy, we'll do it in the future if requested.